File: examples/nodejs/verify_cron_job_actually_ran.js
#!/usr/bin/env node
/**
 * Verify Cron Job Actually Ran
 * Problem: Cron job is scheduled but may not execute
 * Solution: Send explicit ping at start and end of job
 * Documentation: https://deadmanping.com/verify-cron-job-actually-ran
 */

const https = require('https');

const MONITOR_ID = 'YOUR_MONITOR_ID';  // Replace with your actual monitor ID

// Your actual work
async function run() {
  await performBackup();
  
  // Single ping at end - if job fails, ping won't arrive and DeadManPing will alert
  https.request(`https://deadmanping.com/api/ping/${MONITOR_ID}`, { method: 'POST' }).end();
}

run().catch((err) => {
  // If job fails, ping won't arrive - DeadManPing will detect missing ping
  process.exit(1);
});

File: examples/python/verify_cron_job_actually_ran.py
#!/usr/bin/env python3
"""
Verify Cron Job Actually Ran
Problem: Cron job is scheduled but may not execute
Solution: Send explicit ping when job completes successfully
Documentation: https://deadmanping.com/verify-cron-job-actually-ran
"""

import requests

MONITOR_ID = "YOUR_MONITOR_ID"  # Replace with your actual monitor ID

# Your actual work
perform_backup()

# Ping on success - if this doesn't arrive, DeadManPing will alert
requests.post(f"https://deadmanping.com/api/ping/{MONITOR_ID}")
